plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/PartitionTable.java

```diff
@@ -291,6 +291,17 @@ public static Object convert(Object value, Type type)
             // TODO the client sees the bytearray's tostring ouput instead of seeing actual bytes, needs to be fixed.
             return ((ByteBuffer) value).array();
         }
+        if (type instanceof Types.IntegerType || type instanceof Types.DateType) {
+            // date transform applied
+            if (value instanceof Integer) {
+                int actual = (int) value;
+                return (long) actual;
+            }
+            if (value instanceof Long) {
+                return value;
+            }
```

Comment:
Nit: maybe just:
```suggestion
            if (value instanceof Number) {
                return ((Number) value).longValue();
            }
```

Reply:
Please don't use cast to Number. It hides a lot of possible bugs.